### PR TITLE
Fix order of columnState param

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Fix order of `columnState` param in composables.

#### WHY

> [Modifier](https://developer.android.com/reference/kotlin/androidx/compose/ui/Modifier) should appear as the first optional parameter in the parameter list; after all required parameters (except for trailing lambda parameters) but before any other parameters with default values.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScalingLazyColumnState.kt [hunk 1]
```diff
@@ -100,8 +100,8 @@ public class ScalingLazyColumnState(
 
 @Composable
 public fun ScalingLazyColumn(
-    modifier: Modifier = Modifier,
     columnState: ScalingLazyColumnState,
+    modifier: Modifier = Modifier,
```
Comment (luizgrp) [inline compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScalingLazyColumnState.kt:104]: @yschimke are you happy with this change? highlighting this as it's not marked as experimental

Comment (yschimke) [reply compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScalingLazyColumnState.kt:104]: It probably should be.  Can you add?

**Review by yschimke — APPROVED** [review]

File: compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScalingLazyColumnState.kt (hunk 1 above)
Comment (luizgrp) [reply compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScalingLazyColumnState.kt:104]: will add in another PR as I have another branch chained to this one

Comment (luizgrp) [reply compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScalingLazyColumnState.kt:104]: There are loads of classes in compose-layout without the experimental annotation. Should I apply it to all or just this one?

Comment (luizgrp) [reply compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScalingLazyColumnState.kt:104]: https://github.com/google/horologist/pull/837

**Pull request merged**